src/lib/warn-upload-overwrite.js

```diff
@@ -160,7 +182,7 @@ const preUploadDoc = async (db, localDoc) => {
   return Promise.resolve(true);
 };
 
-const postUploadDoc = doc => updateStoredHash(doc._id, getDocHash(doc));
+const postUploadDoc = (doc) => getDocHash(doc).then((docHash) => updateStoredHash(doc._id, docHash));
```

Comment:
This whole file uses `async/await` instead of promises' `then`. Let's not mix things up, please :)